File: lib/models/job_seeker.dart
class JobSeeker {
  final String id;
  final String firstName;
  final String lastName;
  final String? profileImage;
  final bool isMarried;
  final List<String> skills;
  final String location;
  final int expectedSalary;
  final double rating;
  final bool isSmoker;
  final bool hasAddiction;
  final DateTime createdAt;

  JobSeeker({
    required this.id,
    required this.firstName,
    required this.lastName,
    this.profileImage,
    required this.isMarried,
    required this.skills,
    required this.location,
    required this.expectedSalary,
    this.rating = 0.0,
    this.isSmoker = false,
    this.hasAddiction = false,
    DateTime? createdAt,
  }) : createdAt = createdAt ?? DateTime.now();

  String get fullName => '$firstName $lastName';
}

File: models/JobSeeker.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const JobSeeker = sequelize.define('JobSeeker', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  userId: {
    type: DataTypes.INTEGER,
    field: 'user_id',
    references: { model: 'users', key: 'id' }
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false
  },
  age: {
    type: DataTypes.INTEGER
  },
  experience: {
    type: DataTypes.INTEGER,
    defaultValue: 0
  },
  skills: {
    type: DataTypes.JSON,
    defaultValue: []
  },
  expectedSalary: {
    type: DataTypes.BIGINT,
    field: 'expected_salary'
  },
  location: {
    type: DataTypes.STRING
  },
  province: {
    type: DataTypes.STRING
  },
  phoneNumber: {
    type: DataTypes.STRING(11),
    field: 'phone_number'
  },
  description: {
    type: DataTypes.TEXT
  },
  profileImage: {
    type: DataTypes.STRING,
    field: 'profile_image'
  },
  isMarried: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
    field: 'is_married'
  },
  isSmoker: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
    field: 'is_smoker'
  },
  hasAddiction: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
    field: 'has_addiction'
  },
  isActive: {
    type: DataTypes.BOOLEAN,
    defaultValue: true,
    field: 'is_active'
  },
  isApproved: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
    field: 'is_approved'
  }
}, {
  tableName: 'job_seekers'
});

module.exports = JobSeeker;
